services/cases-api/lambdas/updateCase.js

```diff
@@ -48,6 +46,28 @@ export async function main(event) {
       return response.failure(new BadRequestError('invalid [statusType]'));
     }
 
+    let expireHours = undefined;
+
+    /**
+     * If the case is being updated to set the status to 'active:ongoing' or 'active:submitted:viva', set the expiration time in seconds.
+     */
+    /* eslint-disable prettier/prettier */
+    switch (status.type) {
+    case 'active:ongoing':
+      expireHours = CASE_ONGOING_EXPIRATION_HOURS;
+      break;
+    case 'active:submitted:viva':
+      expireHours = CASE_SUBMITTED_EXPIRATION_HOURS;
+      break;
+    }
+    /* eslint-enable prettier/prettier */
+
+    if (expireHours !== undefined) {
+      const newExpirationTime = millisecondsToSeconds(getFutureTimestamp(expireHours));
+      UpdateExpression.push('expirationTime = :newExpirationTime');
+      ExpressionAttributeValues[':newExpirationTime'] = newExpirationTime;
+    }
+
```

Comment:
Hi Max,
At this moment the expire time is only relevant for cases with the provider 'VIVA'

My suggestion is that you extract this logic in a separate lambda and put it under the viva-ms service.

Then make the lambda trigger on status: active:ongoing or active:submitted:viva. See serverless.yml (viva-ms) for more help in setting the event bridge params

I'd be glad if we could replace the switch statement for a function call as well :D

Just poke me if this make no sense.

Reply:
Moved it to its own lambda, and removed the switch statement.